typed_python/PyClassInstance.cpp

```diff
@@ -202,17 +211,28 @@ PyObject* PyClassInstance::pyTernaryUnaryOperatorConcrete(PyObject* rhs, PyObjec
 
 int PyClassInstance::pyInquiryConcrete(const char* op, const char* opErrRep) {
     // op == '__bool__'
-    std::pair<bool, PyObject*> p = callMemberFunction("__bool__", nullptr, nullptr);
+    auto p = callMemberFunction("__bool__");
     if (!p.first) {
-        p = callMemberFunction("__len__", nullptr, nullptr);
+        p = callMemberFunction("__len__");
         // if neither __bool__ nor __len__ is available, return True
         if (!p.first)
             return 1;
     }
     return PyObject_IsTrue(p.second);
 }
 
-std::pair<bool, PyObject*> PyClassInstance::callMemberFunction(const char* name, PyObject* arg0, PyObject* arg1) {
+// try to call user-defined hash method
+// returns -1 if not defined or if it returns an invalid value
+int64_t PyClassInstance::tryCallHashMemberFunction() {
+    auto result = callMemberFunction("__hash__");
```

Comment:
if __hash__ returns `-1` we need to return `-2`.

Reply:
I'll make a test for this case.  It looks like the -1 to -2 transformation is already handled, in the compiled case by tp_hash_to_py_hash, and in the interpreted case by PyInstance::tp_hash.